PayPalNative/src/main/java/com/braintreepayments/api/PayPalNativeClient.java

```diff
@@ -1,49 +1,265 @@
+                        final String pairingId = payPalResponse.getPairingId();
+                        if (payPalResponse.isBillingAgreement()) {
+                            createOrderActions.setBillingAgreementId(pairingId);
+                        } else {
+                            createOrderActions.set(pairingId);
+                        }
+                    }
+                },
+                new OnApprove() {
+                    @Override
+                    public void onApprove(final Approval approval) {
+                        braintreeClient.sendAnalyticsEvent("paypal.native.approved");
+                        handleNativeCheckoutApproval(approval, payPalResponse, callback);
+                    }
+                },
+                new OnCancel() {
+                    @Override
+                    public void onCancel() {
+                        braintreeClient.sendAnalyticsEvent("paypal.native.client_cancel");
+                        callback.onResult(null, new BraintreeException("Canceled"));
+                    }
+                },
+                new OnError() {
+                    @Override
+                    public void onError(final ErrorInfo errorInfo) {
+                        braintreeClient.sendAnalyticsEvent("paypal.native.error");
+                        callback.onResult(null, new BraintreeException(errorInfo.getReason()));
+                    }
+                }
+        );
+    }
+
+    private void handleNativeCheckoutApproval(final Approval approval, final PayPalResponse payPalResponse, final PayPalNativeTokenizeCallback callback) {
+        Uri deepLinkUri;
+        String tokenKey;
+        if (payPalResponse.isBillingAgreement()) {
+            deepLinkUri = Uri.parse(String.format(
+                    "%s://onetouch/v1/success?token=%s&ba_token=%s",
+                    braintreeClient.getReturnUrlScheme(),
+                    approval.getData().getOrderId(),
+                    payPalResponse.getPairingId())); //The approval object will return the returnURL
+            tokenKey = "ba_token";
+        } else {
+            deepLinkUri = Uri.parse(String.format(
+                    "%s://onetouch/v1/success?paymentId=%s&token=%s&PayerID=%s",
+                    braintreeClient.getReturnUrlScheme(),
+                    approval.getData().getPaymentId(),
+                    approval.getData().getOrderId(),
+                    approval.getData().getPayerId()));
+            tokenKey = "token";
+        }
+        try {
+            JSONObject urlResponseData = parseUrlResponseData(deepLinkUri, payPalResponse.getSuccessUrl(), payPalResponse.getApprovalUrl(), tokenKey);
+            PayPalAccount payPalAccount = new PayPalAccount();
+            payPalAccount.setClientMetadataId(payPalResponse.getClientMetadataId());
+            payPalAccount.setSource("paypal-browser"); //TODO: check for valid sources
```

Comment:
It looks like [`BTPayPalNativeTokenizationRequest`](https://github.com/braintree/braintree_ios/blob/pp-checkout-module/Sources/BraintreePayPalNative/Public/BraintreePayPalNative/BTPayPalNativeTokenizationRequest.swift) in the iOS SDK aligns with `PayPalAccount` here, so we might be able to take `source` from the `clientMetadataId` here.

Reply:
@sarahkoop sorry i dont understand, how do i get the `source` from the `clientMetadataId`?

Reply:
Ah I think I was mistaken. We can leave this as a TODO for now.